src/main/java/projectzero/core/UmlClass.java

```diff
@@ -64,13 +76,23 @@ public boolean addMethod(Method method) {
         return true;
     }
 
+    public boolean addRelationship(Relationship relationship) {
+        if (this.relationshipMap.containsKey(relationship.getTo().getName()) || relationship.getTo().getRelationshipMap().containsKey(this.getName())) {
+            return false;
+        }
+
+        this.relationshipMap.put(relationship.getTo().getName(), relationship);
+        return true;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         UmlClass umlClass = (UmlClass) o;
```

Comment:
This line is a little confusing because the names look identical.
It might be easier to understand if it was UmlClass newUmlClass = (UmlClass) o;